userspace/falco/app_actions/process_events.cpp

```diff
@@ -38,20 +40,32 @@ using namespace falco::app;
 //
 // Event processing loop
 //
-application::run_result application::do_inspect(syscall_evt_drop_mgr &sdropmgr,
-				 std::shared_ptr<stats_writer> statsw,
-				 uint64_t duration_to_tot_ns,
-				 uint64_t &num_evts)
+application::run_result application::do_inspect(
+		std::shared_ptr<sinsp> inspector,
+		const std::string& source, // an empty source represents capture mode
+		std::shared_ptr<stats_writer> statsw,
+		syscall_evt_drop_mgr &sdropmgr,
+		bool check_drops_and_timeouts,
+		uint64_t duration_to_tot_ns,
+		uint64_t &num_evts)
 {
 	int32_t rc;
 	sinsp_evt* ev;
 	stats_writer::collector stats_collector(statsw);
 	uint64_t duration_start = 0;
 	uint32_t timeouts_since_last_success_or_msg = 0;
-	std::size_t source_idx;
-	bool source_idx_found = false;
 	token_bucket rate_limiter;
 	bool rate_limiter_enabled = m_state->config->m_notifications_rate > 0;
+	bool source_engine_idx_found = false;
+	bool is_capture_mode = source.empty();
+	bool syscall_source_engine_idx = m_state->sources.at(falco_common::syscall_source)->engine_idx;
```

Comment:
Same as above; i am worried about the new `do_inspect` performances.

Reply:
Same as above. This is out of the event processing loop and is done before entering in it, so this should not affect runtime performance